packages/data-controls/src/index.ts

```diff
@@ -35,45 +37,67 @@ export function apiFetch( request ) {
  * Control for resolving a selector in a registered data store.
  * Alias for the `resolveSelect` built-in control in the `@wordpress/data` package.
  *
- * @param {Array} args Arguments passed without change to the `@wordpress/data` control.
+ * @param storeNameOrDescriptor
+ * @param selectorName
+ * @param args                  Arguments passed without change to the `@wordpress/data` control.
  */
-export function select( ...args ) {
+export function select(
+	storeNameOrDescriptor: string | StoreDescriptor,
+	selectorName: string,
+	...args: any[]
+) {
 	deprecated( '`select` control in `@wordpress/data-controls`', {
 		since: '5.7',
 		alternative: 'built-in `resolveSelect` control in `@wordpress/data`',
 	} );
 
-	return dataControls.resolveSelect( ...args );
```

Comment:
Typescript wasn't happy about this because it's not specific enough.

Reply:
I like it, it's way more predictable that way.